docs/api_docs.go

```diff
@@ -451,11 +498,39 @@ func readASTFile(fset *token.FileSet, filepath string) (*ast.File, error) {
 }
 
 func sanitizeDocString(s string) string {
+	s = strings.Split(s, ";")[0]
 	s = strings.ReplaceAll(s, "\n", " ")
 	s = strings.ReplaceAll(s, "\"", "'")
 	return s
 }
 
+var jsonTagRe = regexp.MustCompile(`json:"(\w+)`)
+
+func fieldNamePrioritizeJSONTag(f *ast.Field) string {
+	if f.Tag != nil {
+		if matches := jsonTagRe.FindStringSubmatch(f.Tag.Value); len(matches) > 0 {
+			return matches[1]
+		}
+	}
+	return f.Names[0].String()
+}
+
+func parseFieldComment(f *ast.Field) (description, example string) {
+	strs := strings.Split(f.Doc.Text(), ";")
+	switch len(strs) {
+	case 1:
+		return strs[0], ""
+	case 2:
+		example = strings.TrimSpace(strs[1])
```

Comment:
Should we check here if the `example` now starts with "e.g."? Maybe something like this:

```
if !strings.HasPrefix(example, "e.g.") {
  log.Errorf("example must start with 'e.g.', got %q", example)
  return strs[0], ""
}
```